### PR TITLE
Inventory Page Implementations - Le Duy An Tran

### PR DESCRIPTION
**Pull request by tranleduyan**
This PR includes:
    - ER-172 Implement Admin Dashboard Inventory Types API Calls
    - ER-133 Inventory Page Layout
    - ER-165 Implement Get All Models API Calls
    - ER-166 Implement Get All Types API Calls
    - ER-134 Implement Mass Delete Type API Calls
    - ER-135 Implement Mass Delete Model API Calls
Features Descriptions:
    This PR delivered the inventory page with the functionality of viewing, selecting, and mass delete selected models and types.
Optimizations:
    Beside features, it also provides an improvement in reducing the amount needed to call APIs in the AddToInventoryPage, thus, minimize the cost in calling APIs. 
Instructions:
    This PR uses implementations in ER-87-LDA-ModelAddition branch in backend. If the branch has not been merged, please use the branch instead of main.

### CODE REVIEW

**Review by amansiid — APPROVED** [review]
The code follows the convention and everything is organized.

File: src/Components/Cards/TypeInventoryCard/TypeInventoryCard.js
Comment (mohammedking11-spu) [inline src/Components/Cards/TypeInventoryCard/TypeInventoryCard.js:1]: Easy to follow as always :)

File: src/Components/Inventory/EquipmentInventory/EquipmentInventory.css
Comment (mohammedking11-spu) [inline src/Components/Inventory/EquipmentInventory/EquipmentInventory.css:1]: Thanks for considering multiple devices screens.

File: src/Constants.js
Comment (mohammedking11-spu) [inline src/Constants.js:1]: Very nice comments explaining each one.

File: src/Components/Cards/EquipmentInventoryCard/EquipmentInventoryCard.js
Comment (mohammedking11-spu) [inline src/Components/Cards/EquipmentInventoryCard/EquipmentInventoryCard.js:1]: I see that you have declared isSelected on line 21, but you have never used its value. VS code is suggesting that we delete it, unless if you are planning on using it in the future 😁

File: src/Pages/AddToInventoryPage/AddToInventoryPage.js
Comment (mohammedking11-spu) [inline src/Pages/AddToInventoryPage/AddToInventoryPage.js:1]: Did you forget to import/include isTypeFetched? According to VS Code, it says that "React Hook useEffect has a missing dependency: 'isTypeFetched'.

File: src/Pages/InventoryPage/InventoryPage.js [hunk 1]
```diff
@@ -1,18 +1,594 @@
...
+         HiMinusCircle, HiPencilAlt, HiPlus, HiSwitchHorizontal, 
+         HiTrash } from 'react-icons/hi';
+//#endregion
+
 // Define InventoryPage Component
-function InventoryPage() {
+function InventoryPage(props) {
+
+  // Extract necessary props - TODO: Implement userRole restriction access
+  const { userRole, schoolId } = props;
```
Comment (mohammedking11-spu) [inline src/Pages/InventoryPage/InventoryPage.js:36]: Is userRole necessary? According to VS Code, userRole is declared but its value never read. Just wanted to make sure 😀

**Review by mohammedking11-spu — APPROVED** [review]
All in all, this gets my approval. As always, code was easy to follow and follows the same format/pattern. Abides with the code conventions that we have agreed upon. Just small things I included in the comments 😁 but other than that, everything looks solid. It works, and it solves what the task was asking about.

File: src/Components/Cards/EquipmentInventoryCard/EquipmentInventoryCard.js
Comment (tranleduyan) [reply src/Components/Cards/EquipmentInventoryCard/EquipmentInventoryCard.js:1]: Yes, it is for the future use when we implement get all equipment! I can add a todo comment to remind me until we can implement it!

File: src/Components/Cards/TypeInventoryCard/TypeInventoryCard.js
Comment (tranleduyan) [reply src/Components/Cards/TypeInventoryCard/TypeInventoryCard.js:1]: Thank you, Mohammed!

File: src/Pages/AddToInventoryPage/AddToInventoryPage.js
Comment (tranleduyan) [reply src/Pages/AddToInventoryPage/AddToInventoryPage.js:1]: Thank you for suggesting me! It is not necessary to use equipmentSection anymore. I should have think of the stateful at first. I have changed it!

File: src/Pages/InventoryPage/InventoryPage.js (hunk 1 above)
Comment (tranleduyan) [reply src/Pages/InventoryPage/InventoryPage.js:36]: Yes, it is for future use when we finish this page! We will use the userRole to restrict access from the student and faculty. Admins are the only ones can go to this page. It is not used for now for convenience.

**Pull request merged**